pkg/security/ldapclient/client.go

```diff
@@ -98,7 +137,7 @@ func (l *ldapClientConfig) Connect() (ldap.Client, error) {
 		return con, nil
 
 	case ldaputil.SchemeLDAPS:
-		return ldap.DialTLS("tcp", l.host, tlsConfig)
+		return ldapDialTLS("tcp", l.host, tlsConfig)
```

Comment:
I would only ask you to provide a `tcpdump` of the communication from the oauth-server pod to an encrypted LDAP server.

We need to be super-careful about our own supplied logic here. While we are testing LDAP IdP, I don't think we test it with a server that would refuse non-TLS connections.

Please see the following test, it might help you set up the IDP: https://github.com/openshift/origin/blob/804b1c878eb827068ba558d8940979e426298103/test/extended/oauth/ldap.go#L19-L20 (you might use this test to set up the server) -

Reply:
https://gist.github.com/alvaroaleman/2e35dcc2640192b8d7e8daabfcfb49be

Top one is with the insecure option set, bottom one without. Both are captured during a login attempt using the IDP